src/main/extension/extension_helper.cpp

```diff
@@ -136,6 +137,44 @@ bool ExtensionHelper::AllowAutoInstall(const string &extension) {
 	return false;
 }
 
+bool ExtensionHelper::CanAutoloadExtension(const string &ext_name) {
+	if (ext_name.empty()) {
+		return false;
+	}
+	for (const auto &ext : AUTOLOADABLE_EXTENSIONS) {
+		if (ext_name == ext) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void ExtensionHelper::AutoLoadExtension(ClientContext &context, const string &extension_name) {
+	auto &dbconfig = DBConfig::GetConfig(context);
+	try {
+		if (dbconfig.options.autoinstall_known_extensions) {
+			ExtensionHelper::InstallExtension(context, extension_name, false,
+			                                  context.config.autoinstall_extension_repo);
+		}
+		ExtensionHelper::LoadExternalExtension(context, extension_name);
+	} catch (Exception &e) {
+		string install_hint;
+
+		if (!dbconfig.options.autoinstall_known_extensions) {
```

Comment:
Should this (ideally) NOT be shown if either settings are locked OR DUCKDB_DISABLE_EXTENSION_LOAD is provided?

Unsure what's the proper way to structure this in code.
Should also this point to the website? `See https://duckdb.org/autoloading for more info`
(Mostly since there is more space there AND that can be updated if we have feedback (while messages are sort of set)